codex-process-feasibility/src/test/java/de/netzwerk_universitaetsmedizin/codex/processes/feasibility/service/DownloadFeasibilityResourcesTest.java

```diff
@@ -0,0 +1,194 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class DownloadFeasibilityResourcesTest {
+
+    @Mock
+    private FhirWebserviceClientProvider clientProvider;
+
+    @Mock
+    private FhirWebserviceClient localWebserviceClient;
+
+    @Mock
+    private TaskHelper taskHelper;
+
+    @Mock
+    private DelegateExecution execution;
+
+    @Mock
+    private Task task;
+
+    @InjectMocks
+    private DownloadFeasibilityResources service;
+
+    @Test
+    public void testDoExecute_NoMeasureReference() {
+        when(execution.getVariable(Mockito.eq("task"))).thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, ConstantsFeasibility.CODESYSTEM_FEASIBILITY,
+                ConstantsFeasibility.CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.empty());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_BundleWithTooFewResultEntries() {
+        final Reference measureRef = new Reference("http://localhost/Measure/id-151003");
+
+        final Bundle measureOnlyBundle = new Bundle(new Enumeration<>(new Bundle.BundleTypeEnumFactory()));
+        measureOnlyBundle.setType(Bundle.BundleType.BATCHRESPONSE);
+
+        when(execution.getVariable(Mockito.eq("task")))
+                .thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, ConstantsFeasibility.CODESYSTEM_FEASIBILITY,
+                ConstantsFeasibility.CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.of(measureRef));
+        when(clientProvider.getLocalWebserviceClient())
+                .thenReturn(localWebserviceClient);
+        when(clientProvider.getRemoteWebserviceClient(Mockito.anyString()))
+                .thenReturn(localWebserviceClient);
+        when(localWebserviceClient.searchWithStrictHandling(Mockito.any(), Mockito.anyMap()))
+                .thenReturn(measureOnlyBundle);
+        when(localWebserviceClient.getBaseUrl()).thenReturn("http://localhost");
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_FirstBundleEntryIsNoMeasure() {
+        final Reference measureRef = new Reference("http://localhost/Measure/id-151003");
+
+        final Bundle.BundleEntryComponent patientEntry = new Bundle.BundleEntryComponent();
+        patientEntry.setId("id-165214");
+        patientEntry.setResource(new Patient().setId("id-170524"));
+
+        final Bundle.BundleEntryComponent measureEntry = new Bundle.BundleEntryComponent();
+        measureEntry.setId("id-151003");
+        measureEntry.setResource(new Measure().setId("id-170418"));
```

Comment:
A non-Measure first entry is sufficient.

Reply:
No it's not. If we were about to test it like this then the `bundle.getEntry().size() < 2` check within the tested class would return true and we would end up in another execution branch.

Reply:
Ok you are right. But please use two Patients to keep things simpler.